### ISSUE TITLE
Upgrade extensions to support EDC `v0.0.1-milestone-8`

### ISSUE THREAD

**Issue by schoenenberg** [sovity/edc-extensions#155]
# Bug Report / Feature request

## Description

_A clear and concise description of the bug._
_If applicable, add screenshots or other information to help explain your problem._

I would like to use these extensions with EDC `v0.0.1-milestone-8`, because of other extensions depending on this version. Therefore I am a bit in a dependency conflict, because these extensions are working fine with `v0.0.1-milestone-7` and the others are working fine with `v0.0.1-milestone-8`. I think in this case the upgrade is preferred over the downgrade..

### Expected Behavior

_A clear and concise description of what you expected to happen._

Just work with `v0.0.1-milestone-8`. No exceptions in the log.

### Observed Behavior

_A clear and concise description of what happened instead._

Compiles without error. But I am getting this Exception in the log:

```
2023-02-28 12:01:27 Initialized IDS Core
Exception in thread "main" java.lang.NoSuchMethodError: 'void org.eclipse.edc.protocol.ids.api.multipart.dispatcher.sender.IdsMultipartSender.<init>(org.eclipse.edc.spi.monitor.Monitor, okhttp3.OkHttpClient, org.eclipse.edc.protocol.ids.spi.service.DynamicAttributeTokenService, com.fasterxml.jackson.databind.ObjectMapper)'
        at de.sovity.extension.broker.BrokerExtension.registerBrokerMessageSenders(BrokerExtension.java:255)
        at de.sovity.extension.broker.BrokerExtension.initialize(BrokerExtension.java:131)
        at org.eclipse.edc.boot.system.injection.lifecycle.InitializePhase.initialize(InitializePhase.java:37)
        at org.eclipse.edc.boot.system.injection.lifecycle.ExtensionLifecycleManager.initialize(ExtensionLifecycleManager.java:61)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at org.eclipse.edc.boot.system.ExtensionLoader.bootServiceExtensions(ExtensionLoader.java:67)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.bootExtensions(BaseRuntime.java:152)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:222)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:87)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:75)
```

## Steps to Reproduce

Basically just upgrade in a sample project the edc-version from `v0.0.1-milestone-7` to `v0.0.1-milestone-8` and see what happens in the log on startup.

## Context Information

_Add any other context about the problem here._

- Environment: MDS Dev-Environment
- Used version:
  - EDC: `v0.0.1-milestone-8`
  - sovity/edc-extensions: v1.3.0
- Time: 2023-02-28 12:01:27 UTC
- Logs: see above
- Parameter: _happens on startup_
- OS: Debian Docker Image
- ...

## Possible Implementation and Work Breakdown

_You already know the root cause of the erroneous state and how to fix it? Feel free to share your thoughts._

I don't know.

**Comment by SebastianOpriel** [comment]
Thank you for your input. Before we discuss further steps, could you elaborate on 
> these extensions are working fine with v0.0.1-milestone-7 and **the others** are working fine with v0.0.1-milestone-8

What **others** do you mean?

**Comment by schoenenberg** [comment]
I mean "the other extensions I am using right now". Those are private _data-plane_ extensions from another company and therefore, not up me to disclose details about them. I hope this is ok for now

**Comment by tmberthold** [comment]
Small update:
With Milestone 8 the `IdsMultipartSender` now needs an `EdcHttpClient` instead of `OkHttpClient` as one of the breaking changes. We are currently working on the upgrade.

**Comment by SebastianOpriel** [comment]
@schoenenberg We plan to release the MS8 upgraded sovity Community Edition in week 20 (from 20th on). In the meantime, you might use the ms8 upgrade branch
